docs/proposals/20230130-go-configuration.md

```diff
@@ -0,0 +1,189 @@
+# Observatorium Golang-based Configuration
+
+* **Owners:**
+  * `@saswatamcode` `@moadz`
+
+* **Other docs:**
+  * None
+
+## TL;DR
+
+This proposal explores the possiblity of using Go as a configuration language and adding Observatorium configuration libraries written in Go that allow end-users to import and customize them directly. It explores the pitfalls of using jsonnet as a front-level interface for Observatorium, and how this currently affects Observatorium installation model.
+
+## Pitfalls of Current Solution
+
+Currently, we primarily use [jsonnet](https://jsonnet.org/) in Observatorium, to compose a "meta-project" library that allows generating configuration for a multi-tenant scalable observability system.
+
+While jsonnet is a commonly used configuration templating language, there are certain issues with how jsonnet acts as a front-level configuration interface, which causes it to not be preferable to end-users.
+
+Templating via jsonnet, while achievable, often results in complex, unreadable code, and due to no doc generation and low IDE support becomes virtually undiscoverable. Codifying and abstracting operational practices and knowledge in jsonnet is difficult and time-consuming, due to the lack of type safety and testing practices. In certain cases, pre-existing upstreams also cannot be used and jsonnet templating needs to be written manually.
+
+Furthermore, for a project like Observatorium, this leads to a complex-to-adopt and messy installation model. An end-user is forced to go through one of the below,
```

Comment:
Or what if the end-user already uses jsonnet in their environment due to wider community adoption (i.e. from Grafana, Prometheus, etc)?

Reply:
That's fine! 

It still means the end-user would have to learn how the Observatorium jsonnet lib is constructed and then spend time navigating and modifying tricky jsonnet to customize it for their pipelines. Maybe spend even more time trying to work observatorium into their jsonnet.

With this, they can have separate easier-to-operate Go code which they can customize to output YAML they need.

In fact, this is what we do downstream with our Observatorium RBAC. We maintain it in Go because it's easier to modify than jsonnet templating: https://github.com/rhobs/configuration/blob/main/configuration/observatorium/rbac.go. 🙂

Reply:
What I mean is that the proposal seems biased toward people that do not already know jsonnet. So saying Observatorium is complex to adopt and has a messy installation model is not a fact, it's relatively to who does it. 

This part of the proposal could be probably rewritten something in the lines of "we assume (and assumptions could be wrong) that there aren't many people out there with enough jsonnet knowledge and for people new to it its usage is complex and messy".

Reply:
I mean, we are jsonnet users, we deal with it on a daily basis, and know how to use it. And still, find it difficult and tricky to maintain, adopt and scale with it! I'm not sure if this is a biased view?

Reply:
I think even wider community does not adopt jsonnet as fast as we envisioned.

So I assume @douglascamata you just want to rephrase existing to something like 
```suggestion
Furthermore, for a project like Observatorium, this leads to a complex-to-adopt installation model, especially for those not familiar with jsonnet. An end-user is forced to go through one of the below,
```